src/vcpkg/base/files.cpp

```diff
@@ -2653,8 +2701,43 @@ namespace vcpkg
                                std::error_code& ec) override
         {
 #if defined(_WIN32)
-            return stdfs::copy_file(
-                to_stdfs_path(source), to_stdfs_path(destination), convert_copy_options(options), ec);
+            DWORD last_error;
+            auto wide_source = Strings::to_utf16(source.native());
+            auto wide_destination = Strings::to_utf16(destination.native());
+            if (options != CopyOptions::overwrite_existing)
+            {
+                if (::CopyFileW(wide_source.c_str(), wide_destination.c_str(), TRUE))
+                {
+                    ec.clear();
+                    return true;
+                }
+
+                last_error = GetLastError();
+                if (last_error != ERROR_FILE_EXISTS || options == CopyOptions::none)
+                {
+                    ec.assign(static_cast<int>(last_error), std::system_category());
+                    return false;
+                }
+
+                // open handles to both files in exclusive mode to implement the equivalent() check
+                FileHandle source_handle(wide_source.c_str(), FILE_READ_DATA, 0, OPEN_EXISTING, 0, ec);
+                if (ec)
+                {
+                    return false;
+                }
+                FileHandle destination_handle(wide_destination.c_str(), FILE_WRITE_DATA, 0, OPEN_EXISTING, 0, ec);
+                return false;
+            }
+
+            if (::CopyFileW(wide_source.c_str(), wide_destination.c_str(), FALSE))
+            {
+                // ec is already cleared here by opening destination_handle
```

Comment:
Needs `ec.clear()`. "Opening destination_handle" isn't on this code path any longer.